cli/compose/convert/service.go

```diff
@@ -435,3 +447,20 @@ func convertDeployMode(mode string, replicas *uint64) (swarm.ServiceMode, error)
 	}
 	return serviceMode, nil
 }
+
+func convertCredentialSpec(source string) (*swarm.CredentialSpec, error) {
+	if source == "" {
+		return nil, nil
+	}
+
+	credentialSpec := swarm.CredentialSpec{}
+	switch {
+	case strings.HasPrefix(source, "file://"):
```

Comment:
I think it's (mostly) fine on the command line because having lots of flags that are basically the same is gross... however providing a strongly typed object which we are representing in the yaml format is much more desirable. 

Otherwise we have to rely on the daemon to parse these strings.